src/Servers/IIS/IIS/src/Core/IO/AsyncIOEngine.cs

```diff
@@ -35,13 +37,85 @@ public ValueTask<int> ReadAsync(Memory<byte> memory)
     }
 
     public ValueTask<int> WriteAsync(ReadOnlySequence<byte> data)
+    {
+        if (SegmentsOverChunksLimit(data))
+        {
+            return WriteDataOverChunksLimit(data);
+        }
+
+        return WriteDataAsync(data);
+    }
+
+    private ValueTask<int> WriteDataAsync(in ReadOnlySequence<byte> data)
     {
         var write = GetWriteOperation();
         write.Initialize(_handler, data);
         Run(write);
         return new ValueTask<int>(write, 0);
     }
 
+    /*
+    * In case the number of chunks is bigger than responseMaxChunks we need to make multiple calls 
+    * to the native api https://docs.microsoft.com/en-us/iis/web-development-reference/native-code-api-reference/ihttpresponse-writeentitychunks-method  
+    * Despite the documentation states that feeding the function with more than 65535 chunks will cause the function to throw an exception, 
+    * it actually seems that 65534 is the maximum number of chunks allowed.
+    * Also, there seems to be a problem when slicing a ReadOnlySequence on segment borders tracked here https://github.com/dotnet/runtime/issues/67607
+    * That's why we only allow 65533 chunks.  
+    */
```

Comment:
```suggestion
    // In case the number of chunks is bigger than responseMaxChunks we need to make multiple calls
    // to the native api https://docs.microsoft.com/en-us/iis/web-development-reference/native-code-api-reference/ihttpresponse-writeentitychunks-method
    // Despite the documentation states that feeding the function with more than 65535 chunks will cause the function to throw an exception,
    // it actually seems that 65534 is the maximum number of chunks allowed.
    // Also, there seems to be a problem when slicing a ReadOnlySequence on segment borders tracked here https://github.com/dotnet/runtime/issues/67607
    // That's why we only allow 65533 chunks.
```